### PR TITLE
Adding license header to repo files

### PR DESCRIPTION
**Pull request by hitenshKharva**
### Description
- Adding License Header to all the missing copyright files

### Issues Resolved
Issue: [4140](https://github.com/opensearch-project/opensearch-build/issues/4140)

### Notes
This issue was created to make sure the PR [4118](https://github.com/opensearch-project/opensearch-build/pull/4118) for issue [3079](https://github.com/opensearch-project/opensearch-build/issues/3079) does not fail the CI workflow.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peterzhuamazon — APPROVED** [review]
Thanks @hitenshKharva .

Awaiting @gaiksaya approval.

Thanks.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
```
Comment (gaiksaya) [inline gradle/wrapper/gradle-wrapper.properties:7]: Anything in `gradle/wrapper` folder can be ignored.

**Review by gaiksaya — CHANGES_REQUESTED** [review]
Hi @hitenshKharva 

Overall looks good. One nit: Can you make the comments uniform with `#` instead of `\\`
Thanks!

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (hitenshKharva) [reply gradle/wrapper/gradle-wrapper.properties:7]: Do you want me to ignore the folder or make the comments in gradle/wrapper uniform?

Comment (gaiksaya) [reply gradle/wrapper/gradle-wrapper.properties:7]: Ignore `gradle/wrapper` and make other license comments uniform by changing `\\` to `#`

File: jenkins/check-for-build.jenkinsfile [hunk 2]
```diff
@@ -1,3 +1,10 @@
+//  Copyright OpenSearch Contributors
+//  SPDX-License-Identifier: Apache-2.0
+
+//  The OpenSearch Contributors require contributions made to
+//  this file be licensed under the Apache-2.0 license or a
+//  compatible open source license.
+
```
Comment (gaiksaya) [inline jenkins/check-for-build.jenkinsfile:7]:
```suggestion
#  Copyright OpenSearch Contributors
#  SPDX-License-Identifier: Apache-2.0

#  The OpenSearch Contributors require contributions made to
#  this file be licensed under the Apache-2.0 license or a
#  compatible open source license.

```

File: jenkins/cross-cluster-replication/perf-test.jenkinsfile [hunk 3]
```diff
@@ -1,3 +1,10 @@
+//  Copyright OpenSearch Contributors
+//  SPDX-License-Identifier: Apache-2.0
+
+//  The OpenSearch Contributors require contributions made to
+//  this file be licensed under the Apache-2.0 license or a
+//  compatible open source license.
```
Comment (gaiksaya) [inline jenkins/cross-cluster-replication/perf-test.jenkinsfile:6]:
```suggestion
#  Copyright OpenSearch Contributors
#  SPDX-License-Identifier: Apache-2.0

#  The OpenSearch Contributors require contributions made to
#  this file be licensed under the Apache-2.0 license or a
#  compatible open source license.
```

**Review by gaiksaya — COMMENTED** [review]
These are the comment changes I was talking about in various files you made changes to

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (hitenshKharva) [reply gradle/wrapper/gradle-wrapper.properties]: Added license comments as discussed over Slack.

File: jenkins/cross-cluster-replication/perf-test.jenkinsfile (hunk 3 above)
Comment (gaiksaya) [reply jenkins/cross-cluster-replication/perf-test.jenkinsfile]: Synced up offline. Jenkinsfile is basically groovy so had to use
```
/*
*
*
*/
```

**Pull request merged**